std/format.d

```diff
@@ -988,7 +988,7 @@ struct FormatSpec(Char)
                 const widthOrArgIndex = parse!uint(tmp);
                 enforceFmt(tmp.length,
                     text("Incorrect format specifier %", trailing[i .. $]));
-                i = tmp.ptr - trailing.ptr;
+                i = arrayPtrDiff(tmp, trailing);
```

Comment:
Alternatively, we could use a more general ptrValue safe function:
`i = tmp.ptrValue - trailing.ptrValue;`
See https://github.com/dlang/druntime/pull/1590#issuecomment-225712106